integration-tests/bats/sql.bats

```diff
@@ -39,6 +39,83 @@ teardown() {
     teardown_common
 }
 
+@test "sql: errors do not write incomplete rows" {
```

Comment:
Would like to see equivalents of these tests in the engine unit tests as well, it's pretty important